transformer_engine/pytorch/tensor/quantized_tensor.py

```diff
@@ -286,21 +286,6 @@ def detach(self) -> QuantizedTensor:
             f"{self.__class__.__name__} class does not implement detach function"
         )
 
-    def update_usage(
-        self,
-        rowwise_usage: Optional[bool] = None,
-        columnwise_usage: Optional[bool] = None,
-    ):
-        """Indicate to the tensor how it is going to be used
-
-        This enables optimizations to memory usage in some cases
-        where forward and backward passes use the tensor in
-        different directions.
-        """
-        raise NotImplementedError(
-            f"{self.__class__.__name__} class does not implement update_usage function"
-        )
-
```

Comment:
This is hinting that our class design is not quite right. This function established an API that we could use generically, and now that it's gone it just so happens that the subclasses implement a function with the same name. The right answer is probably to add a base class for the mixin classes, which would also make the `is_quantized_tensor` function kind of redundant. If we go that way, we should also remember that the internal tensors are a performance optimization, and it should be valid to implement a `QuantizedTensor` without an internal class (and indeed, that should be preferred when prototyping experimental quantization recipes instead of the current trend where we get million-line PRs implementing all optimizations at once).

Reply:
Ok, I caved in and created the base class for the *Base tensors. Will need to do some experiments though how that impacts things and clean up stuff more.

Reply:
In the meantime converted to draft.